tests/src/Kernel/DiscoveryFetcherTest.php

```diff
@@ -0,0 +1,342 @@
+    $file = dirname(__DIR__, 2) . '/fixtures/discovery.mimetypes.xml';
+    $this->xml = file_get_contents($file);
+    $this->httpClientGet = fn(...$args) => new Response(
+      200,
+      [],
+      $this->xml,
+    );
+
+    $client = $this->createMock(Client::class);
+    $client->method('get')
+      ->willReturnCallback(
+        function (...$args): Response {
+          $this->httpClientGetCalls[] = $args;
+          return ($this->httpClientGet)(...$args);
+        },
+      );
+    $this->container->set('http_client', $client);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function assertPostConditions(): void {
+    parent::assertPostConditions();
+    $this->assertNoFurtherLogMessages();
+  }
+
+  /**
+   * Tests a successful call to ->getDiscovery().
+   */
+  public function testGetDiscovery(): void {
+    $fetcher = $this->getFetcher();
+    $discovery = $fetcher->getDiscovery();
+    $this->assertSame(
+      'http://collabora.test:9980/browser/61cf2b4/cool.html?',
+      $discovery->getWopiClientURL(),
+    );
+    $this->assertSame(
+      [
+        [
+          'https://localhost:9980/hosting/discovery',
+          ['verify' => TRUE],
+        ],
+      ],
+      $this->httpClientGetCalls,
+    );
+  }
+
+  /**
+   * Tests that subsequent calls are cached.
+   */
+  public function testGetDiscoveryIsCached(): void {
+    $fetcher = $this->getFetcher();
+    $load_cache = fn () => \Drupal::cache()->get(CollaboraDiscoveryFetcher::DEFAULT_CID);
+    $this->assertFalse($load_cache());
+
+    $fetcher->getDiscovery();
+    $this->assertCount(1, $this->httpClientGetCalls);
+
+    $cache_record = $load_cache();
+    $this->assertNotFalse($cache_record);
+    $this->assertSame(['config:collabora_online.settings'], $cache_record->tags);
+
+    $fetcher->getDiscovery();
+    $this->assertCount(1, $this->httpClientGetCalls);
+
+    /** @var \Drupal\Core\Cache\CacheTagsInvalidatorInterface $invalidator */
+    $invalidator = \Drupal::service(CacheTagsInvalidatorInterface::class);
+    $invalidator->invalidateTags(['config:collabora_online.settings']);
+
+    $fetcher->getDiscovery();
+    $this->assertCount(2, $this->httpClientGetCalls);
+
+    $this->config('collabora_online.settings')
+      ->set('cool.discovery_cache_ttl', 12345)
+      ->save();
+
+    $fetcher->getDiscovery();
+    $this->assertCount(3, $this->httpClientGetCalls);
+    $cache_record = $load_cache();
+    $this->assertNotFalse($cache_record);
+    $this->assertSame(12345, $cache_record->expire - $cache_record->created);
+
+    $this->mockRequestTime = $this->mockRequestTime->add(new \DateInterval('PT12330S'));
+
+    $this->assertSame(12330, $this->mockRequestTime->getTimestamp() - $cache_record->created);
+    $this->assertNotFalse($load_cache());
+
+    $fetcher->getDiscovery();
+    $this->assertCount(3, $this->httpClientGetCalls);
+
+    $this->mockRequestTime = $this->mockRequestTime->add(new \DateInterval('PT30S'));
```

Comment:
```suggestion
    $this->mockRequestTime = $this->mockRequestTime->add(\DateInterval::createFromDateString('30 seconds'));
```
I think it helps the reader